### PR TITLE
Tree service docs

### LINKED ISSUE #527: Describe tree-service usage in docs
Prepare document which briefly describes how S3 Gateway uses tree service.

### PR DESCRIPTION
**Pull request by masterSplinter01**
Closes #527

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
 
 - [Configuration](./docs/configuration.md)
 - [NeoFS AuthMate](./docs/authmate.md)
+- [NeoFS Tree service](./docs/tree_service.md)
```
Comment (KirillovDenis) [inline README.md:91]: Did you forget to push `./docs/tree_service.md`?

File: config/config.yaml [hunk 2]
```diff
@@ -1,102 +1,102 @@
...
-
-# Address to listen and TLS
-listen_address: 0.0.0.0:8084
-tls:
-  cert_file: /path/to/cert
-  key_file: /path/to/key
-
-logger:
-  level: debug
+## Wallet address, path to the wallet must be set as cli parameter or environment variable
```
Comment (KirillovDenis) [inline config/config.yaml:1]: Why did you comment all config values?

File: docs/tree_service.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Each node keeps one of the types of data as a set of **key-value pairs**:
+* Bucket settings: lock configuration and versioning mode 
+* Bucket tagging
+* Object tagging
+* Object metadata: OID, name, creation time, system metadata
+* Object locking settings
+* Active multipart upload info
+
+Some data takes up a lot of memory, so we store it in NeoFS nodes as an object with payload.
```
Comment (KirillovDenis) [inline docs/tree_service.md:14]: The same as previous

File: docs/tree_service.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Tree service
+
+To get objects' metadata and system information, the S3 GW makes requests to the Tree service.
```
[the remark this answers is not in the document]
Comment (EliChin) [reply docs/tree_service.md:3]: I like comma here, cause it's a god practice to place comma in structures like "To do something..., we will..."

**Review by EliChin — APPROVED** [review]

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**